app/helpers/publish/subject_helper.rb

```diff
@@ -4,13 +4,13 @@ module Publish
   module SubjectHelper
     def primary_form_options(subjects = primary_subjects)
       subjects.map do |subject|
-        PrimarySubjectInput.new(1 + subject.subject_code.to_i, subject.subject_name)
```

Comment:
We should be doing following the principle of least astonishment otherwise we are overburdening with unnecessary.

Reply:
Indeed